ui/ui-components/pages/model/[modelId]/source/[sourceId]/edit.tsx

```diff
@@ -86,8 +86,8 @@ const Page: NextPage<Props> = ({
   }, [preview]);
 
   const mappingColumn = useMemo(
-    () => Object.keys(source.mapping)[0] ?? previewColumns?.[0],
```

Comment:
I don't recall why this was added in the previews change, so I restored it to the previous version. @krishnaglick do you recall the reason?